modules/core/src/main/java/org/apache/ignite/internal/processors/cache/checker/processor/PartitionReconciliationProcessor.java

```diff
@@ -0,0 +1,600 @@
+
+            skippedEntries
+                .computeIfAbsent(cacheName, k -> new HashMap<>())
+                .computeIfAbsent(partId, l -> new HashSet<>())
+                .addAll(data);
+        }
+    }
+
+    /**
+     *
+     */
+    private Repair repair(
+        long sesId,
+        UUID workloadChainId,
+        String cacheName,
+        int partId,
+        Map<KeyCacheObject, Map<UUID, GridCacheVersion>> notResolvingConflicts,
+        Map<KeyCacheObject, Map<UUID, VersionedValue>> actualKeys,
+        int repairAttempts
+    ) {
+        Map<KeyCacheObject, Map<UUID, VersionedValue>> res = new HashMap<>();
+
+        for (KeyCacheObject key : notResolvingConflicts.keySet()) {
+            Map<UUID, VersionedValue> versionedByNodes = actualKeys.get(key);
+            if (versionedByNodes != null)
+                res.put(key, versionedByNodes);
+        }
+
+        return new Repair(sesId, workloadChainId, cacheName, partId, res, repairAttempts);
+    }
+
+    /**
+     * Adds processed keys to the total result.
+     */
+    private void addToPrintResult(
+        String cacheName,
+        int partId,
+        Map<KeyCacheObject, Map<UUID, GridCacheVersion>> conflicts,
+        Map<KeyCacheObject, Map<UUID, VersionedValue>> actualKeys
+    ) {
+        CacheObjectContext ctx = ignite.cachex(cacheName).context().cacheObjectContext();
+
+        synchronized (inconsistentKeys) {
+            try {
+                inconsistentKeys.computeIfAbsent(cacheName, k -> new HashMap<>())
+                    .computeIfAbsent(partId, k -> new ArrayList<>())
+                    .addAll(mapPartitionReconciliation(conflicts, actualKeys, ctx));
+            }
+            catch (IgniteCheckedException e) {
+                log.error("Broken key can't be added to result. ", e);
+            }
+        }
+    }
+
+    /**
+     * Add data to print result.
+     *
+     * @param cacheName Cache name.
+     * @param partId Partition Id.
+     * @param repairedKeys Repaired keys.
+     */
+    private void addToPrintResult(
+        String cacheName,
+        int partId,
+        Map<PartitionKeyVersion, RepairMeta> repairedKeys
+    ) {
+        CacheObjectContext ctx = ignite.cachex(cacheName).context().cacheObjectContext();
+
+        synchronized (inconsistentKeys) {
+            try {
+                List<PartitionReconciliationDataRowMeta> res = new ArrayList<>();
+
+                for (Map.Entry<PartitionKeyVersion, RepairMeta> entry : repairedKeys.entrySet()) {
+                    Map<UUID, PartitionReconciliationValueMeta> valMap = new HashMap<>();
+
+                    for (Map.Entry<UUID, VersionedValue> uuidBasedEntry : entry.getValue().getPreviousValue().entrySet()) {
+                        Optional<CacheObject> cacheObjOpt = Optional.ofNullable(uuidBasedEntry.getValue().value());
+
+                        valMap.put(
+                            uuidBasedEntry.getKey(),
+                            cacheObjOpt.isPresent() ?
```

Comment:
Please, do not abuse by multiline ternary operators. Use if-else instead in order to improve readability.